telemetry/telemetrybuffer.go

```diff
@@ -303,8 +303,15 @@ func ReadConfigFile(filePath string) (TelemetryConfig, error) {
 
 // ConnectToTelemetryService - Attempt to spawn telemetry process if it's not already running.
 func (tb *TelemetryBuffer) ConnectToTelemetryService(telemetryNumRetries, telemetryWaitTimeInMilliseconds int) {
-	path, dir := getTelemetryServiceDirectory()
+
+	path, dir, fileExists := getTelemetryServiceDirectory()
 	args := []string{"-d", dir}
+
+	if fileExists {
```

Comment:
it should not be added here.. its after connect failure and before starting new telemetry service process

Reply:
It was supposed to be !fileExists. Does that resolve this?

Reply:
Moved check for binary to match request.